azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/ConfigurationParser.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.maven.webapp.parser;
+
+import com.microsoft.azure.management.appservice.JavaVersion;
+import com.microsoft.azure.management.appservice.RuntimeStack;
+import com.microsoft.azure.management.appservice.WebContainer;
+import com.microsoft.azure.maven.webapp.AbstractWebAppMojo;
+import com.microsoft.azure.maven.webapp.WebAppConfiguration;
+import com.microsoft.azure.maven.webapp.configuration.OperatingSystemEnum;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.codehaus.plexus.util.StringUtils;
+import java.util.List;
+
+public abstract class ConfigurationParser {
+    protected final AbstractWebAppMojo mojo;
+
+    protected ConfigurationParser(final AbstractWebAppMojo mojo) {
+        this.mojo = mojo;
+    }
+
+    protected String getAppName() throws MojoExecutionException {
+        if (StringUtils.isEmpty(mojo.getAppName())) {
```

Comment:
Possible to check the name is illegal or not?

Reply:
Will add a regex expression to validate it.